pybamm/discretisations/discretisation.py

```diff
@@ -147,7 +156,26 @@ def process_model(self, model, inplace=True):
 
         model_disc.bcs = self.bcs
 
-        # Process initial condtions
+        self.external_variables = model.external_variables
+        # find where external variables begin in state vector
```

Comment:
maybe add a comment here saying something about how the external vars are always appended to the end of the state vector, so you only care where they start?